modules/Reminders/Reminder.php

```diff
@@ -292,84 +303,339 @@ public static function addNotifications(jsAlerts $alert, $checkDecline = true)
-                            $alert->addAlert($app_strings['MSG_JS_ALERT_MTG_REMINDER_MEETING'], $meetingName,
-                                $app_strings['MSG_JS_ALERT_MTG_REMINDER_TIME'] . $time,
-                                $app_strings['MSG_JS_ALERT_MTG_REMINDER_LOC'] . $location .
-                                $description .
-                                $instructions,
-                                $timeStart - strtotime($alertDateTimeNow),
-                                $url
-                            );
-                        }
+        // Pre-calculated values in seconds
+        $_24hours = 24;
+        $_1hour = 3600;
+
+        /**
+         * In case the user changes the duration dom dropdown
+         * We need to ensure that we know how many calendar slots there can be in a day.
+         * This is used to limit the query the default should be 96 possible calendar slots in a day.
+         */
+        $_smallestEventPeriod = array_reduce(
+            array_keys($app_list_strings['duration_dom']),
+            function ($carry, $item) {
+
+                if (empty($carry) && empty($item)) {
+                    $carry = $item;
+                } elseif (empty($carry) && !empty($item) && is_numeric($item)) {
+                    $carry = $item;
+                } elseif ($item < $carry) {
+                    $carry = $item;
+                }
+
+                return $carry;
+            }
+        );
+
+        // By default: There are 96 possible slots in 24hours that a reminder can refer to.
+
+        $_maxEventsInADay = ceil(($_1hour / $_smallestEventPeriod) * $_24hours);
+        $userNOW = $timedate->getNow(true)->sub(new DateInterval('PT'. $app_list_strings['reminder_max_time'].'S'))->asDb(false);
+        /**
+         * Optimisation:
+         * Use a single query with a limit to get all the information needed.
+         *
+         * Starting with the current user id with the following table relationships
+         * $current_user->id: Reminder Invitees -< Reminders -< Reminder -< [Meeting/Call] -< [Meeting/Call]_user
+         *
+         * Returns Meetings_<field> or Calls_<field> depending if the reminder_related_event_module is set to Meetings
+         * or Calls.
+         *
+         * It limited by the total number of slots that can be in a calendar day. Since the user is only interested in
+         * alerts which are a day ahead. It is reasonable to assume that a user will not need more than the 96 by
+         * default.
+         *
+         * This means we can optimise the how many results need to be processed each time the user requests the meeting
+         *
+         */
+        // Generate Related to (parent)queries
+        $callParentSelect = PHP_EOL;
+        $callParentJoin = PHP_EOL;
+        $meetingParentSelect = PHP_EOL;
+        $meetingParentJoin = PHP_EOL;
+
+
+        switch ($sugar_config['dbconfig']['db_type']) {
+            case "mssql":
```

Comment:
I have update my query so that the db manager handles the db specific statements.